src/daffodilDebugger/debugger.ts

```diff
@@ -58,12 +58,16 @@ async function getTDMLConfig(
     config.schema.path = ''
 
     config.tdmlConfig.path =
-      config.tdmlConfig.path ||
-      (await vscode.commands.executeCommand(
-        'extension.dfdl-debug.getValidatedTDMLPath'
-      ))
+      config.tdmlConfig.path == '${AskForValidatedTDMLPath}' ||
+      !config.tdmlConfig.path
+        ? await vscode.commands.executeCommand(
+            'extension.dfdl-debug.getValidatedTDMLPath'
+          )
+        : config.tdmlConfig.path
```

Comment:
This is a case where a ternary is harder to read than an if statement. The `config.tdmlConfig.path == '${AskForValidatedTDMLPath}'` is the main culprit here (and should be ===).

``` 
if (!config.tdmlConfig.path || config.tdmlConfig.path === '${AskForValidatedTDMLPath}')
{
  config.tdmlConfig.path = await vscode.commands.executeCommand('extension.dfdl-debug.getValidatedTDMLPath')
}
```